File: LeetCode/DataStruct/JS/Leetcode804-唯一摩斯密码词.js
var uniqueMorseRepresentations = function(words) {
    const pos = [".-","-...","-.-.","-..",".","..-.","--.","....","..",
    ".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-",
    "...-",".--","-..-","-.--","--.."];
    let st = new Set();
    for(const w of words) {
        let s = '';
        for(const x of w) {
            s += pos[x.charCodeAt(0) - 'a'.charCodeAt(0)];
        }
        st.add(s);
    }
    return st.size;
};

File: LeetCode/DataStruct/Python/Leetcode804-唯一摩斯密码词.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        pos = [
            ".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..",
            ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.",
            "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."
        ]
        st = set()
        for w in words:
            s = ''
            for v in w:
                s += pos[ord(v) - ord('a')]
            st.add(s)
        return len(st)
